CONTRIBUTING.rst

```diff
@@ -0,0 +1,201 @@
+tool, add new users and can do so even faster with your help!
+
+Following the guidelines will help us work together as efficiently as
+possible. When we all work with a common understanding, we can make sure
+that issues are addressed quickly, suggested changes can be easily
+assessed and pull requests can be finalised painlessly.
+
+All contributions are welcome, some possible suggestions include:
+
+-  bug reports (make a new issue and use the template please :D)
+-  feature requests (make a new issue and use the template please :D)
+-  pull requests (make a pull request and use the template please :D)
+-  tutorials (or support questions which, once solved, result in a new
+   tutorial :D)
+-  improving the documentation
+-  blog posts
+
+Please don’t use the repository to have discussions about the results.
+Such discussions are scientific and generally belong in the scientific
+literature, not in a development repository.
+
+
+Ground Rules
+************
+
+As a contributor, it is vital that we all follow a few conventions:
+
+-  Be welcoming to newcomers and encourage diverse new contributors from
+   all backgrounds. See the :ref:`code_of_conduct`.
+-  Create issues for changes and enhancements, this ensures that
+   everyone in the community has a chance to comment
+-  Use our ``.editorconfig`` file (see `Editor
+   Config <#editor-config>`__)
+-  Ensure that you pass all the tests before making a pull request
+-  Avoid pushing directly to master, all changes should come via pull
+   requests
+
+
+Setup
+*****
+
+Editor Config
+~~~~~~~~~~~~~
+
+The repository contains a ``.editorconfig`` file. This ensures that all
+of our text editors behave the same way and avoids spurious changes
+simply due to differing whitespace or end of line rules.
+
+Many editors have built in support for ``Editorconfig`` but some require
+a plugin. To work out if your editor requires a plugin, please check
+https://editorconfig.org/.
+
+
+Getting started
+***************
+
+Your First Contribution
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The development methodology for openscm makes heavy use of ``git``,
+``make``, virtual environments and test driven development. If you
+aren’t familiar with any of these terms it might be helpful to spend
+some time getting up to speed on these technologies. Some helpful
+resources (the longest take about 5 hours to work through):
+
+-  `git intro by software
+   carpentry <https://swcarpentry.github.io/git-novice/>`__
+-  `making a pull request <http://makeapullrequest.com/>`__
+-  `my first pull request <http://www.firsttimersonly.com/>`__
+-  `tests intro by software
+   carpentry <https://v4.software-carpentry.org/test/index.html>`__ and
+   `getting started with
+   mocking <https://semaphoreci.com/community/tutorials/getting-started-with-mocking-in-python>`__
+-  `make intro by software
+   carpentry <https://swcarpentry.github.io/make-novice/>`__
+-  `venv <https://docs.python.org/3/library/venv.html>`__
```

Comment:
Add references to 

- `Continuous integration (CI) <https://docs.travis-ci.com/user/for-beginners/>`_
    - we use `Travis CI <https://travis-ci.com/>`_ for our CI but there are a number of good providers
- `Jupyter Notebooks <https://medium.com/codingthesmartway-com-blog/getting-started-with-jupyter-notebook-for-python-4e7082bd5d46>`_
    - we'd recommend simply installing ``jupyter`` (``conda install jupyter``) in your virtual environment
- `Sphinx <http://www.sphinx-doc.org/en/master/>`_